src/backend/sev/unattested_launch.rs

```diff
@@ -46,7 +47,24 @@ pub fn launch(sock: UnixStream) -> Result<()> {
     let msr = from_reader(&sock).context("failed to deserialize expected measurement packet")?;
     assert!(matches!(msr, Message::Measurement(_)));
 
-    let secret_packet = Message::Secret(None);
+    let secret_packet = if let Message::Measurement(msr) = msr {
+        let measurement: ::sev::launch::Measurement = msr.measurement;
+
+        let session = unsafe { session.mock_verify(measurement).context("verify failed")? };
+
+        let ct_vec = vec![0u8, 1u8, 2u8, 3u8, 4u8, 5u8];
+        let mut ct_enc = Vec::new();
+        into_writer(&Bytes::from(ct_vec), &mut ct_enc).context("failed to encode secret")?;
+
+        let secret = session
+            .secret(::sev::launch::HeaderFlags::default(), &ct_enc)
+            .context("gen_secret failed")?;
+
+        Message::Secret(Some(secret))
+    } else {
+        Message::Secret(None)
+    };
```

Comment:
This `else` is unreachable. The `assert!` statement above it guarantees the first branch will always be taken.

One suggestion here would be to effectively replace the `assert!(maches!(...));`

```rust
let msr = match msr {
    Message::Measurement(m) => m,
    _ => panic!("expected measurement packet"),
};

let secret_packet = {
    let session = unsafe { session.mock_verify(msr).context("verify failed")? };
    
    // [...]
};
```

Reply:
Thanks! Now it looks much better!